mario/pipelines/app_barsdetection.py

```diff
@@ -0,0 +1,128 @@
+from metaflow import FlowSpec, Parameter, kubernetes, secrets, step, trigger
+
+
+@trigger(event='app-barsdetection')
+class AppBarsdetection(FlowSpec):
+    """Run a transcript through app-barsdetection"""
+
+    guid = Parameter('guid', help='GUID of the transcript to process')
+    mmif = Parameter('mmif', help='GUID of the transcript to process', default=None)
+
+    @secrets(sources=['CLAMS-SonyCi-API', 'CLAMS-chowda-secret'])
+    @kubernetes(
+        image='ghcr.io/wgbh-mla/chowda:main',
+        persistent_volume_claims={'media-pvc': '/m'},
+    )
+    @step
+    def start(self):
+        """Download the media file"""
+        from os.path import join
+        from subprocess import run
+        from urllib.request import urlretrieve
+
+        from chowda.db import engine
+        from chowda.models import MediaFile
+        from requests import post
+        from sonyci import SonyCi
+        from sqlmodel import Session, select
+
+        # get SonyCi Asset ID
+        with Session(engine) as db:
+            media_file = db.exec(
+                select(MediaFile).where(MediaFile.guid == self.guid)
+            ).one()
+            self.asset_id = media_file.assets[0].id
+
+        assert self.asset_id, f'No asset found for {self.guid}'
+        print(f'Found asset {self.asset_id}')
+        self.filename = join('/m', self.guid + '.mp4')
+
+        # get mmif
+        self.input_mmif = self.mmif
+        if not self.mmif:
+            print('No mmif provided, downloading from clams')
+            self.input_mmif = post(
+                'http://fastclam/source',
+                json={'files': ['video:' + self.filename]},
+            ).json()
+        print('Got mmif')
+        print(self.input_mmif)
+        # Download the media file
+        # run(f'ci download {self.asset_id} -o {join("/m", self.guid)}')
+        ci = SonyCi(**SonyCi.from_env())
+        self.asset = ci.get(f'assets/{self.asset_id}')
+
+        url = self.asset['proxyUrl']
+        print('Downloading file')
+        urlretrieve(url, self.filename)
```

Comment:
`self.filename` is set to `{guid}.mp4` but the Sony Ci Asset name is often something slightly different. This difference might not matter functionally, but if the filename is part of logs that we are using for reporting and/or debugging, it might be worth it to keep the Sony Ci Asset name as the filename for traceability. Just a suggestion